### ISSUE TITLE
Execute Shell doesn't work on Chrome or Firefox

### ISSUE THREAD

**Issue by hdep** [rancher/rancher#10643]
using firefox 57.0.2 or Chrome Version 63.0.3239.84 (Build officiel) (64 bits)
the remote shell isn't showing up properly for any container. Please see the following screenshots : 

On Firefox :
![firefox](https://user-images.githubusercontent.com/12408502/34265278-b2cdae04-e675-11e7-84a4-b0799ea78d2d.png)
On Chrome : 
![chrome](https://user-images.githubusercontent.com/12408502/34265279-b2e5f61c-e675-11e7-9fe7-99695579a61a.png)

I test it on Microsoft Edge 41.16299.15.0 it works.

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.6.12` Cattle: `v0.183.30` UI: `v1.6.30` |
|Access|`ldap` `admin`|
|Orchestration|`Cattle`|
|Route|`container.ports`|

**Comment by loganhz** [comment]
Hi @hdep , thanks for your reporting.
I can not reproduce this issue on Mac and Windows 10.
What OS are you using? Is there any useful info in Chrome console log?

**Comment by hdep** [comment]
Hello,

I'm on windows 10 too.
it seems that my Chrome wasn't up-to-date so I updated to  Version 63.0.3239.108 (Build officiel) (64 bits) and now it works.

I still have the same issue with firefox. 
In the console logs I only have : 

```
Subscribe ping (projectId=1a29405, sockId=1)
ui-bbbfef6b68d1d7b48e4f9f261d4dfe49.js:97:14999
You did not pass the `focus` argument in `Terminal.prototype.open()`.
The `focus` argument now defaults to `true` but starting with xterm.js 3.0 it will default to `false`.
vendor-77489dadbb0709acf44882f0dde4b110.js:22:16391
Subscribe ping (projectId=1a29405, sockId=1)
ui-bbbfef6b68d1d7b48e4f9f261d4dfe49.js:97:14999
```
I'm not sure this help.

**Comment by loganhz** [comment]
Thanks @hdep . PR has been submitted. This issue will be fixed in next release.

**Comment by MichelDiz** [comment]
I'm having this same issue. But in all browsers. The installation is from scratch. It was working in version v1.6.12, but I decided to switch to Latest (1.6.13) and it was like this. No way to see logs or get into the shell.

**Comment by loganhz** [comment]
Hi @MichelDiz ,  this issue happens in shell only. I guess it shows `Disconnected` in your case. Do you have a reverse proxy setup in front of your rancher server ? Can you provide the `rancher/agent` log when it happens, please?

**Comment by MichelDiz** [comment]
No, I usually use Traefik, but this time I'm not using anything.

Actually, I do not even know what's going on anymore. After this behave I deleted the machine (VM) to engage new tests, the system started to behave (new one) strangely.

I deleted all and install a new Docker (Docker Tools) then install a version of Rancher (tried a few different) and he simply does not want to install the agent anymore. And the agent get stuck. I can't stop it. Just if I run these commands below.

docker stop $(docker ps -a -q)
docker rm $(docker ps -a -q)

I think I will have to format my computer because I have already done everything I can and what I understand about Computing and it does not solve. Not even the old "reboot".

----

> Also before i was doing a schema where I put Rancher: Server on a virtual machine from my PC and the agent on a Google Cloud machine. But it did the same behave confirmed as below (this data below is from the test from now on a local machine VM) and simply did not appear the Host in the list.
>I found it to be some problem with NoIP or even my router. However, I did all route, curl, and ping tests. Everything was handy.
> 
> In order for my machine to be visible to the Google cloud I simply made a "NoIP" scheme and used an address of them. I opened the ports (forwarding), everything as I always did and worked before. But it did not turn positive this time :/

```bash
docker run --rm --privileged -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/rancher:/var/lib/rancher ranche
r/agent:v1.2.7 http://192.168.99.100:8080/v1/scripts/04AA20C7095A67787344:1483142400000:5a0rffbHSMJfEPS36hVOGnMNek
Unable to find image 'rancher/agent:v1.2.7' locally
v1.2.7: Pulling from rancher/agent
b3e1c725a85f: Already exists
6a710864a9fc: Already exists
d0ac3b234321: Already exists
87f567b5cf58: Already exists
063e24b217c4: Already exists
d0a3f58caef0: Already exists
16914729cfd3: Already exists
fb770dddb34d: Pull complete
0d8557fbdb4c: Pull complete
Digest: sha256:fcd7075d1aee2b532319846f70dc7ed0acfb0d12045cfc82ff995f282ef14631
Status: Downloaded newer image for rancher/agent:v1.2.7

INFO: Running Agent Registration Process, CATTLE_URL=http://192.168.99.100:8080/v1
INFO: Attempting to connect to: http://192.168.99.100:8080/v1
INFO: http://192.168.99.100:8080/v1 is accessible
INFO: Inspecting host capabilities
INFO: Boot2Docker: true
INFO: Host writable: false
INFO: Token: xxxxxxxx
INFO: Running registration
INFO: Printing Environment
INFO: ENV: CATTLE_ACCESS_KEY=7A628BBB687D3FDB7E35
INFO: ENV: CATTLE_HOME=/var/lib/cattle
INFO: ENV: CATTLE_REGISTRATION_ACCESS_KEY=registrationToken
INFO: ENV: CATTLE_REGISTRATION_SECRET_KEY=xxxxxxx
INFO: ENV: CATTLE_SECRET_KEY=xxxxxxx
INFO: ENV: CATTLE_URL=http://192.168.99.100:8080/v1
INFO: ENV: DETECTED_CATTLE_AGENT_IP=172.17.0.1
INFO: ENV: RANCHER_AGENT_IMAGE=rancher/agent:v1.2.7
INFO: Launched Rancher Agent: 059df38e61c07d45f0d45cc345bc85947a4c9cad1ddfd38815aa37c9fce91a08
```

**Comment by deniseschannon** [comment]
@hdep We aren't able to reproduce on the latest versions of Chrome (63.0.3239.132 (Official Build) (64-bit))/Firefox (57.0.4).

We put in a fix in the UI in v1.6.14-rc2, but I am unable to definitively verify it since I don't have the issue on the most recent versions. 

If you still face this issue with v1.6.14+ and you're on the latest browsers, please re-open.
